src/workerd/server/workerd-api.c++

```diff
@@ -266,6 +264,12 @@ struct WorkerdApi::Impl final {
         jsgIsolate(v8System, Configuration(*this), kj::mv(observerParam), kj::mv(createParams)),
         memoryCacheProvider(memoryCacheProvider),
         pythonConfig(pythonConfig) {
+    // maybeOwnedModuleRegistry is only set when using the
+    // new module registry implementation. When that is the
+    // case we also need to tell JSG.
+    if (maybeOwnedModuleRegistry != kj::none) {
+      jsgIsolate.setUsingNewModuleRegistry();
```

Comment:
Honestly kinda seems like an odd way to do this.
maybeOwnedModuleRegistry is available before constructing jsgIsolate.
usingNewModuleRegistry can more logically be defined as a boolean argument to jsgIsolate's constructor

Reply:
I *really* dislike boolean arguments to constructors and this follows the pattern we have with other flag-controlled behaviors.

Reply:
Honestly I don't see how this is better